tfutils/base.py

```diff
@@ -718,16 +764,35 @@ def test(sess,
             How frequently to save intermediate results captured during test
             None means no intermediate saving will be saved
     """
-    coord, threads = start_queues(sess)
-    dbinterface.start_time_step = time.time()
-    valid_results_summary = run_targets_dict(sess,
-                                             valid_targets,
-                                             save_intermediate_freq=save_intermediate_freq,
-                                             dbinterface=dbinterface,
-                                             validation_only=True)
-    dbinterface.sync_with_host()
-    stop_queues(sess, queues, coord, threads)
-    return valid_results_summary, dbinterface.outrecs
+    # Collect args in a dict of lists
+    test_args = {
+        'sess': sess,
+        'queues': queues,
+        'dbinterface': dbinterface,
+        'validation_targets': validation_targets,
+        'save_intermediate_freq': save_intermediate_freq}
+
+    _ttargs = [{key: value[i] for (key, value) in test_args.items()}
```

Comment:
because sess is always the same across the list, just break it out and pass one session everywhere